File: v2/benchmark/tests/integration/test_hello_world.py
"""
Unit tests for Hello World function
"""

from hello_world import hello_world


def test_hello_world_default():
    """Test default parameter"""
    assert hello_world() == "Hello, World!"
    print("✓ Default parameter test passed")


def test_hello_world_custom_name():
    """Test with custom name"""
    assert hello_world("Claude") == "Hello, Claude!"
    print("✓ Custom name test passed")


def test_hello_world_empty_string():
    """Test with empty string"""
    assert hello_world("") == "Hello, !"
    print("✓ Empty string test passed")


if __name__ == "__main__":
    print("Running Python Hello World Tests:\n")
    
    test_hello_world_default()
    test_hello_world_custom_name()
    test_hello_world_empty_string()
    
    print("\nAll tests completed!")

File: v2/benchmark/archive/old-files/test_hello_world.js
/**
 * Unit tests for Hello World function
 */

import { helloWorld } from './hello_world.js';

// Simple test framework
function test(description, testFunc) {
  try {
    testFunc();
    console.log(`✓ ${description}`);
  } catch (error) {
    console.log(`✗ ${description}`);
    console.error(`  Error: ${error.message}`);
  }
}

function assert(condition, message) {
  if (!condition) {
    throw new Error(message || 'Assertion failed');
  }
}

// Run tests
console.log('Running JavaScript Hello World Tests:\n');

test('should return "Hello, World!" with no parameter', () => {
  assert(helloWorld() === 'Hello, World!', 'Default parameter not working');
});

test('should return custom greeting with name parameter', () => {
  assert(helloWorld('Claude') === 'Hello, Claude!', 'Custom name not working');
});

test('should handle empty string', () => {
  assert(helloWorld('') === 'Hello, !', 'Empty string not handled');
});

console.log('\nAll tests completed!');